### PR TITLE
Display your strong Pokemon

### PR DESCRIPTION
**Pull request by Nordiii**
As some people request I added this functionality again to display your stronger pokemon!

#553

### CODE REVIEW

**Comment by GDlike** [comment]
+    <Setting Name="AmountOfPokemonToDisplayOnStart" Type="System.Int32" Scope="User"> 
+      <Value Profile="(Default)">999</Value> 

Wow amazing we can with this, setup to show all our pokemon, I really like this lol

Thank you!

**Comment by yamashi** [comment]
It appears to have conflicts with the branch, please fix before I can merge.
Thanks

**Comment by Nordiii** [comment]
Currently dealing with the unwanted logger lines in logic, will fix the conflict after this!

**Comment by KiriharaSenpai** [comment]
is there any news about it

**Comment by Nordiii** [comment]
I'm still stuck at the Event to keep their code clean sadly I get an error which i still cannot identify

**Comment by solenrus** [comment]
gl, I want this, was exactly my thought on how this should work, with a global settings and stuff

File: PoGo.NecroBot.Logic/Event/DisplayHighestsPokemonEvent.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PoGo.NecroBot.Logic.Event
+{
+    public class DisplayHighestsPokemonEvent : IEvent
+    {
+        public String sortetBy;
+        //PokemonData | CP | IV | Level
+        public List<Tuple<PokemonData,int,double,double>> pokemonList;
```
Comment (yamashi) [inline PoGo.NecroBot.Logic/Event/DisplayHighestsPokemonEvent.cs]: Upper case

File: PoGo.NecroBot.Logic/Event/DisplayHighestsPokemonEvent.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PoGo.NecroBot.Logic.Event
+{
+    public class DisplayHighestsPokemonEvent : IEvent
+    {
+        public String sortetBy;
```
Comment (yamashi) [inline PoGo.NecroBot.Logic/Event/DisplayHighestsPokemonEvent.cs]: SortedBy

File: PoGo.NecroBot.Logic/Tasks/LogBestPokemonTask.cs [hunk 3]
```diff
@@ -13,21 +17,51 @@ public static class LogBestPokemonTask
...
+            {
+
+                machine.Fire(               
+                    new DisplayHighestsPokemonEvent
+                    {
+                        sortetBy = "Cp",
+                        pokemonList = pokemonPairedWithStatsCP
+                    });
+            }
+            catch (RuntimeBinderException e)
```
Comment (yamashi) [inline PoGo.NecroBot.Logic/Tasks/LogBestPokemonTask.cs]: This exception means the StatisticAggregator is not handling the event properly.

**Comment by Nordiii** [comment]
I hope its okay now :)!

File: PoGo.NecroBot.Logic/Event/DisplayHighestsPokemonEvent.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PoGo.NecroBot.Logic.Event
+{
+    public class DisplayHighestsPokemonEvent : IEvent
+    {
+        public String SortetBy;
```
Comment (yamashi) [inline PoGo.NecroBot.Logic/Event/DisplayHighestsPokemonEvent.cs]: Sorted*

**Comment by enerai** [comment]
@Nordiii Could it be possible to tell how the bot can show more strongest Pokemons (more than 10---the current number in setting)

**Pull request merged**